mapproxy/test/system/test_cache_geopackage.py

```diff
@@ -0,0 +1,309 @@
+        self.cache = GeopackageCache(os.path.join(self.cache_dir, 'tmp.geopackage'),
+                                     configuration.grids.get('GLOBAL_GEODETIC'),
+                                     self.table_name)
+
+    def teardown(self):
+        if self.cache:
+            self.cache.cleanup()
+        TileCacheTestBase.teardown(self)
+
+    def test_get_map_cached(self):
+        prepare_env(test_config, 'cache_geopackage.yaml')
+        create_app(test_config)
+        SystemTest.setup(self)
+        self.common_map_req = WMS111MapRequest(url='/service?', param=dict(service='WMS',
+                                                                           version='1.1.1', bbox='-180,-80,0,0',
+                                                                           width='200', height='200',
+                                                                           layers='gpkg', srs='EPSG:4326',
+                                                                           format='image/png',
+                                                                           styles='', request='GetMap'))
+        resp = self.app.get(self.common_map_req)
+        eq_(resp.content_type, 'image/png')
+        data = BytesIO(resp.body)
+        assert is_png(data)
+
+    def test_get_map_uncached(self):
+        prepare_env(test_config, 'cache_geopackage.yaml')
+        create_app(test_config)
+        SystemTest.setup(self)
+        self.common_map_req = WMS111MapRequest(url='/service?', param=dict(service='WMS',
+                                                                           version='1.1.1', bbox='-180,-80,0,0',
+                                                                           width='200', height='200',
+                                                                           layers='gpkg', srs='EPSG:4326',
+                                                                           format='image/png',
+                                                                           styles='', request='GetMap'))
+        assert os.path.exists(os.path.join(test_config['base_dir'], 'cache.gpkg')) # already created on startup
+
+        self.common_map_req.params.bbox = '-180,0,0,80'
+        serv = MockServ(port=42423)
+        serv.expects('/tiles/01/000/000/000/000/000/001.png')
+        serv.returns(create_tmp_image((256, 256)))
+        with serv:
+            resp = self.app.get(self.common_map_req)
+            eq_(resp.content_type, 'image/png')
+            data = BytesIO(resp.body)
+            assert is_png(data)
+
+        # now cached
+        resp = self.app.get(self.common_map_req)
+        eq_(resp.content_type, 'image/png')
+        data = BytesIO(resp.body)
+        assert is_png(data)
+
+    def test_bad_config_geopackage_srs(self):
+        error_msg = None
+        try:
+            prepare_env(test_config, 'cache_geopackage_srs.yaml')
+            create_app(test_config)
+        except ValueError as ve:
```

Comment:
I think system tests are a bit overkill for this. I would create a few unit tests where you create GeopackageCache objects with varying tile grids.